From: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
Subject: Port to the new gtk_box_new() instead of gtk_hbox_new().

Index: indicator-datetime/src/indicator-datetime.c
===================================================================
--- indicator-datetime.orig/src/indicator-datetime.c	2012-01-12 23:59:50.815675283 +0100
+++ indicator-datetime/src/indicator-datetime.c	2012-01-13 00:02:24.083668449 +0100
@@ -34,6 +34,7 @@
 #include <glib-object.h>
 #include <glib/gi18n-lib.h>
 #include <gio/gio.h>
+#include <gtk/gtk.h>
 
 /* Indicator Stuff */
 #include <libindicator/indicator.h>
@@ -1261,7 +1262,12 @@
 
 	mi_data->gmi = gtk_menu_item_new();
 
+#if GTK_CHECK_VERSION(3, 2, 0)
+	GtkWidget * hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 4);
+	gtk_box_set_homogeneous (GTK_BOX(hbox), FALSE);
+#else
 	GtkWidget * hbox = gtk_hbox_new(FALSE, 4);
+#endif
 
 	/* Icon, probably someone's face or avatar on an IM */
 	mi_data->icon = gtk_image_new();
@@ -1472,7 +1478,12 @@
 	gtk_check_menu_item_set_active(GTK_CHECK_MENU_ITEM(mi_data->gmi),
 		dbusmenu_menuitem_property_get_bool(newitem, TIMEZONE_MENUITEM_PROP_RADIO));
 
+#if GTK_CHECK_VERSION(3, 2, 0)
+	GtkWidget * hbox = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 4);
+	gtk_box_set_homogeneous (GTK_BOX(hbox), FALSE);
+#else
 	GtkWidget * hbox = gtk_hbox_new(FALSE, 4);
+#endif
 
   	/* Label, probably a username, chat room or mailbox name */
 	mi_data->label = gtk_label_new("");
