=== modified file 'src/timezone-completion.c'
--- a/src/timezone-completion.c	2011-09-06 12:47:37 +0000
+++ b/src/timezone-completion.c	2011-10-05 15:47:06 +0000
@@ -355,7 +355,6 @@
   gchar * locale = get_locale ();
   gchar * url = g_strdup_printf (GEONAME_URL, escaped, version, locale);
   g_free (locale);
-  g_free (version);
   g_free (escaped);
 
   GFile * file =  g_file_new_for_uri (url);

